packages/datepicker/src/use-date-picker-offset.ts

```diff
@@ -23,8 +27,12 @@ export const useDatePickerOffsetPropGetters: DPUseDatePickerOffsetPropGetters =
         { disabled, onClick, ...rest }: DPPropsGetterConfig = {},
       ) => {
         const nextDate = getNextOffsetDate(state.offsetDate, offsetValue);
+        const disabledDueToMaxDateAndMonthOffset =
+          !!offsetValue.months &&
+          maxDateAndAfter(maxDate, nextDate) &&
+          isAfterMaxMonth(nextDate.getMonth(), maxDate);
```

Comment:
We can't do this only based on the `offsetValue.months` also we can't set the offset date more that the max/minDate.
So at the minimum we should set it on the edge and this logic is not present.